core/src/main/java/hudson/ProxyConfiguration.java

```diff
@@ -157,13 +157,11 @@ public String getUserName() {
         return userName;
     }
 
-//    This method is public, if it was public only for jelly, then should make it private (or inline contents)
-//    Have left public, as can't tell if anyone else is using from plugins
     /**
-     * @return the password in plain text
+     * @return the proxy password
      */
-    public String getPassword() {
-        return Secret.toString(secretPassword);
+    public Secret getPassword() {
```

Comment:
This breaks compatibility but I don't know of a better way of doing this,
I've ensured that JCasC won't break https://github.com/jenkinsci/configuration-as-code-plugin/pull/769

If this change isn't done, the change functionally works but introduces a security risk in that the password would be exported in plain text.

This change has a compatibility risk but is more secure

Reply:
From what I can tell, Secret seems to migrate automatically from String. The main backward incompatibility would be if anything calls this method directly and not via Jelly or similar.

Reply:
Binary compatibility is broken here for an user. There is no automatic type conversion in Java, not sure what @jvz means here. Migration in stapler?

Reply:
The secret class has a converter that converts strings into secrets automatically.
For a user this change is individual to the best of my knowledge

Reply:
We really need to retain binary compatibility here somehow

Reply:
I did that before with the `getSecretPassword` additional getter which wasn't ideal but @daniel-beck suggested reverting and doing the change as it shouldn't impact anyone?
https://github.com/jenkinsci/jenkins/pull/3935#issuecomment-481440872

open to suggestions though

Reply:
@oleg-nenashev We really don't; I've been unable to identify any plugins using this API.